Sources/Fuzzilli/Modules/NetworkSync.swift

```diff
@@ -104,12 +104,16 @@ class Connection {
     /// Pending outgoing data. Must only be accessed on this connection's dispatch queue.
     private var sendQueue: [Data] = []
     
-    init(socket: Int32, handler: MessageHandler) {
+    init(socket: libsocket.socket_t, handler: MessageHandler) {
         self.socket = socket
         self.handler = handler
         self.queue = DispatchQueue(label: "Socket \(socket)")
-        
+
+#if os(Windows)
+        self.readSource = DispatchSource.makeReadSource(handle: HANDLE(bitPattern: UInt(socket))!, queue: self.queue)
```

Comment:
Just for my understanding, why is the `HANDLE(bitPattern: UInt(socket))!` required here? Shouldn't `socket` already be a HANDLE on Windows? Or is `SOCKET` != `HANDLE` on Windows?

Reply:
`SOCKET` and `HANDLE` are distinct types on Windows.  The reason is that `HANDLE` normally is something vended by the kernel.  `SOCKET` is vended by WinSock, which is just a library.  I do agree with you that the special handling here is ugly.

Reply:
Cool, thanks for the explanation, that makes sense and I'm fine with the code as is! Feel free to add your reply as a comment to it if you want though :)